File: tabuada.py
while True:
    try:
        n = int(input("Qual número deseja ver a tabuada? "))
        print(f"\nTabuada do {n}:")
        for i in range(1, 11):
            print(f"{n} × {i:2} = {n*i:3}")
        print()
        
        continuar = input("Deseja ver outra tabuada? (1 = sim / 0 = não): ")
        if continuar == "0":
            print("Programa encerrado.")
            break
        elif continuar != "1":
            print("Opção inválida. Encerrando...")
            break
            
    except:
        print("Digite um número inteiro válido.\n")

File: tabuada.c
#include <stdio.h>

int main() {
    int num, continuar;
    
    do {
        printf("Qual numero deseja ver a tabuada? ");
        scanf("%d", &num);
        
        printf("\nTabuada do %d:\n", num);
        for (int i = 1; i <= 10; i++) {
            printf("%2d x %2d = %3d\n", num, i, num * i);
        }
        printf("\n");
        
        printf("Deseja ver outra tabuada? (1 = sim / 0 = nao): ");
        scanf("%d", &continuar);
        
    } while (continuar == 1);
    
    printf("Programa encerrado.\n");
    return 0;
}
